lib/perl/Genome/Model/ReferenceAlignment/Command/GenotypeMicroarrayConcordance.pm

```diff
@@ -54,53 +67,47 @@ sub execute {
     my $seqdict = $ref_seq_build->get_sequence_dictionary('sam',$ref_seq_build->species_name,$self->picard_version);
     $self->_seqdict($seqdict);
 
-    my @data;
-    my @instrument_data = $self->build->instrument_data;
-    unless (@instrument_data) {
-        $self->error_message('Found no instrument data assigned to build: '. $self->build->__display_name__);
-        die($self->error_message);
-    }
-    for my $instrument_data (@instrument_data) {
-        my $qc_build = $instrument_data->lane_qc_build;
-        unless ($qc_build) {
-            $self->error_message('Failed to find lane qc build for instrument data : '. $instrument_data->__display_name__);
-            die($self->error_message);
-        }
-        
-        my $lane_qc_vcf = $self->resolve_lane_qc_vcf($qc_build);
-        my ($microarray_vcf, $genotype_sample) = $self->resolve_genotype_microarray_vcf_and_sample($qc_build);
-        
-        my $intersect_vcf = Genome::Sys->create_temp_file_path(
-            Genome::Utility::Text::sanitize_string_for_filesystem($genotype_sample->name) .'_x_'. $instrument_data->id .'.vcf'
-        );
-        my $intersect_cmd = Genome::Model::Tools::BedTools::Intersect->create(
-            input_file_a => $lane_qc_vcf,
-            input_file_a_format => 'bed',
-            input_file_b => $microarray_vcf,
-            output_file => $intersect_vcf,
-            header => 1,
-            use_version => $self->bedtools_version,
-        );
-        unless ($intersect_cmd) {
-            $self->error_message('Failed to create bedtools intersect!');
+    my $microarray_build;
+    if ($self->microarray_build) {
+        $microarray_build = $self->microarray_build;
+        # Validate the microarray_build refseq matches the refalign refseq
+        if ($microarray_build->reference_sequence_build_id ne $ref_seq_build->id) {
+            $self->error_message('Inconsistent reference sequence builds!');
             die($self->error_message);
         }
-        unless ($intersect_cmd->execute) {
-            $self->error_message('Failed to execute bedtools intersect!');
-            die($self->error_message);
+    }
+
+    my @data;
+    for my $build ($self->resolve_builds) {
+        my $build_vcf = $self->resolve_snvs_vcf_for_build($build);
+        unless ($microarray_build) {
+            $microarray_build = $build->genotype_microarray_build;
+            unless ($microarray_build) {
+                $self->error_message('Failed to find a Microarray build.  None exists as input to SNVs build.');
+                die($self->error_message);
```

Comment:
Since the microarray model is an input to the reference alignment model, I assumed compatibility checks would happen at that time.  If that's not the case, then perhaps we do want to check the reference sequence build for consistency.